bundler/spec/dependabot/bundler/update_checker_spec.rb

```diff
@@ -218,6 +221,47 @@
       it { is_expected.to eq(Gem::Version.new("1.9.0")) }
     end
 
+    context "with a private rubygems source", :bundler_v2_only do
+      let(:dependency_files) { project_dependency_files("bundler2/specified_source") }
+
+      let(:requirements) do
+        [{
+          file: "Gemfile",
+          requirement: ">= 0",
+          groups: [],
+          source: { type: "rubygems" }
+        }]
+      end
+      let(:registry_url) { "https://repo.fury.io/greysteil/" }
+      let(:gemfury_business_url) do
+        "https://repo.fury.io/greysteil/api/v1/dependencies?gems=business"
+      end
+      before do
+        # We only need to stub out the version callout since it would
+        # otherwise call out to the internet in a shell command
+        allow(Dependabot::Bundler::NativeHelpers).
+          to receive(:run_bundler_subprocess).
+          with({
+                 bundler_version: "2",
```

Comment:
Is this bundler version set in the stub the only difference between test version? WDYT think about just setting the expectation using something like:

```
bundler_version: bundler_2_available? ? "2" : "1"
```

We don't have the fixture swapping wired up based on `SUITE_NAME` but when we do, the gemfiles uses should be automagically pulled from the right folder, so it might save some test duplication.

Reply:
The fixtures are also different between these tests, but I re-checked and it turns out they don't need to be, so let's go with your suggestion 💯